drivers/sensor/tdk/icm40627/icm40627.c

```diff
@@ -0,0 +1,649 @@
+
+	return 0;
+}
+
+static int icm40627_sample_fetch_gyro(const struct device *dev)
+{
+	const struct icm40627_config *cfg = dev->config;
+	struct icm40627_data *data = dev->data;
+	uint8_t buffer[GYRO_DATA_SIZE];
+
+	int res = cfg->bus_io->read(&cfg->bus, REG_GYRO_DATA_X1, buffer, GYRO_DATA_SIZE);
+
+	if (res) {
+		LOG_ERR("read gyro data failed");
+		return res;
+	}
+
+	data->gyro_x = (int16_t)sys_get_be16(&buffer[0]);
+	data->gyro_y = (int16_t)sys_get_be16(&buffer[2]);
+	data->gyro_z = (int16_t)sys_get_be16(&buffer[4]);
+
+	return 0;
+}
+
+static int icm40627_sample_fetch_temp(const struct device *dev)
+{
+	const struct icm40627_config *cfg = dev->config;
+	struct icm40627_data *data = dev->data;
+	uint8_t buffer[TEMP_DATA_SIZE];
+
+	int res = cfg->bus_io->read(&cfg->bus, REG_TEMP_DATA1, buffer, TEMP_DATA_SIZE);
+
+	if (res) {
+		LOG_ERR("read temp data failed");
+		return res;
+	}
+
+	data->temp = (int16_t)sys_get_be16(&buffer[0]);
+
+	return 0;
+}
+
+static int icm40627_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	uint8_t status;
+	const struct icm40627_config *cfg = dev->config;
+
+	icm40627_lock(dev);
+
+	int res = cfg->bus_io->read(&cfg->bus, REG_INT_STATUS, &status, 1);
+
+	if (res) {
+		LOG_ERR("read INT_STATUS failed");
+		goto cleanup;
+	}
+
+	if (!FIELD_GET(BIT_INT_STATUS_DATA_RDY_INT, status)) {
+		res = -EBUSY;
+		goto cleanup;
+	}
+
+	switch (chan) {
+	case SENSOR_CHAN_ALL:
+		res = icm40627_sample_fetch_accel(dev);
+		if (res) {
+			break;
+		}
+		res = icm40627_sample_fetch_gyro(dev);
+		if (res) {
+			break;
+		}
+		res = icm40627_sample_fetch_temp(dev);
+		break;
+	case SENSOR_CHAN_ACCEL_XYZ:
+	case SENSOR_CHAN_ACCEL_X:
+	case SENSOR_CHAN_ACCEL_Y:
+	case SENSOR_CHAN_ACCEL_Z:
+		res = icm40627_sample_fetch_accel(dev);
+		break;
+	case SENSOR_CHAN_GYRO_XYZ:
+	case SENSOR_CHAN_GYRO_X:
+	case SENSOR_CHAN_GYRO_Y:
+	case SENSOR_CHAN_GYRO_Z:
+		res = icm40627_sample_fetch_gyro(dev);
+		break;
+	case SENSOR_CHAN_DIE_TEMP:
+		res = icm40627_sample_fetch_temp(dev);
+		break;
+	default:
+		res = -ENOTSUP;
+		break;
+	}
+
+cleanup:
+	icm40627_unlock(dev);
+	return res;
+}
+
+static int icm40627_attr_set(const struct device *dev, enum sensor_channel chan,
+			     enum sensor_attribute attr, const struct sensor_value *val)
+{
+	int res = 0;
+	struct icm40627_data *data = dev->data;
+
+	__ASSERT_NO_MSG(val != NULL);
+
+	icm40627_lock(dev);
+
+	switch (chan) {
+	case SENSOR_CHAN_ACCEL_X:
+	case SENSOR_CHAN_ACCEL_Y:
+	case SENSOR_CHAN_ACCEL_Z:
+	case SENSOR_CHAN_ACCEL_XYZ:
+		if (attr == SENSOR_ATTR_SAMPLING_FREQUENCY) {
+			res = icm40627_set_accel_odr(dev, data->accel_hz);
+
+			if (res) {
+				LOG_ERR("Incorrect sampling value");
+			} else {
+				data->accel_hz = val->val1;
+			}
+		} else if (attr == SENSOR_ATTR_FULL_SCALE) {
+			res = icm40627_set_accel_fs(dev, data->accel_fs);
+
+			if (res) {
+				LOG_ERR("Incorrect fullscale value");
+			} else {
+				data->accel_fs = val->val1;
```

Comment:
accel_fs seems to be in G, `sensor.h` specifies that we should be using SI units, expect `val` to be in m/s^2.